### ISSUE TITLE
add php thumb options

### ISSUE THREAD

**Issue by eighthday** [alanpich/tveasyupload#8]
it would really handy if you could set a height and width that could be used by phpthumbof to generate a correctly sized image.

**Comment by neoneddy** [comment]
This can be done via the output.. the point of this is to be dead easy.,.. Alan has an add-on that you might like better called Image+

Here is how I use this TV to crop / scale the output.
`<img src="[[+tv.base-image:phpthumbof=`&w=300&h=250&zc=1`:default=`http://placehold.it/150x150&text=FPO`]]">`

This is nice because the idea here is to give your client less opportunity to break stuff.

**Comment by eighthday** [comment]
Sure, but the problem with outputting in the front end like that is it can really effect page load times.

**Comment by neoneddy** [comment]
Only on first load, it cached after that. We can also do on the fly retina versions this way. One image, many resolutions.

Sent from my rotary telephone
I can be reached at 763-639-3584

**Comment by eighthday** [comment]
I know, but if you make a small change to a page with lots of images, the next visitor will get a long page load - wouldn't it be nice if the source image was the size you actually needed?

**Comment by neoneddy** [comment]
I'm not sure there are many ways to independently cache the the images and not content. 

I think a way to solve you problem would be to trigger a plugin on save that primes the cache of that page and then the site when you're done.

I see your point and its an issue I've had to address with clients. Usually we're seeing a 2-3 sec time to first byte on a gallery page. Then 100ms after that.

I use a link checker script right now after I launch a site or do any major editing.

Sent from my rotary telephone
I can be reached at 763-639-3584

**Comment by eighthday** [comment]
or a tv that generates an image to the correct dimensions, when it is set. 

there is actually refreshCache http://modx.com/extras/package/refreshcache, but it is not very client friendly.

**Comment by neoneddy** [comment]
But for this use case, I'll take a single tv and use it at 3 different sizes not counting retina/mobile resolutions.

Thumbnail in index listing
Facebook meta image
Carousel thumbnail 
Image in carousel
Image on page

These are all ways I use this, I don't want the image baked out, I like a high res reference version.

If I recall direct resize on Evo did something like you're wanting in the tinymce, it would scale and replace the image.

Sent from my rotary telephone
I can be reached at 763-639-3584

**Comment by eighthday** [comment]
sure, in you case you might not need it, which is why it should be optional. The way I do it is I just ask my clients to upload the correct ratio and then use phpthumb to generate the largest size used throughout the site. I then use adaptive images to handle smaller image sizes at different breakpoints.

**Comment by alanpich** [comment]
So the main bit your proposing is to generate the phpthumb cache on TV save rather than first call on page? Would the 'raw' image be kept in the filesystem, or just the scaled version?

**Comment by eighthday** [comment]
Exactly, that would be very helpful - there would be no need to keep the original size image.

**Comment by mdehaan** [comment]
Is there any progress on the development of these feature requests?

**Comment by neoneddy** [comment]
I don't see the development needing to be done :-)  It works fine as-is.
